cpp/tests/io/parquet_experimental_reader_test.cpp

```diff
@@ -123,13 +123,15 @@ auto create_parquet_with_stats()
   cudf::io::parquet_writer_options out_opts =
     cudf::io::parquet_writer_options::builder(cudf::io::sink_info{&buffer}, expected)
       .metadata(std::move(expected_metadata))
-      .row_group_size_rows(5000)
-      .max_page_size_rows(1000)
+      .row_group_size_rows(page_size_for_ordered_tests)
+      .max_page_size_rows(page_size_for_ordered_tests / 5)
+      .compression(cudf::io::compression_type::NONE)
+      .dictionary_policy(cudf::io::dictionary_policy::ALWAYS)
       .stats_level(cudf::io::statistics_freq::STATISTICS_COLUMN);
 
   if constexpr (NumTableConcats > 1) {
-    out_opts.set_row_group_size_rows(20000);
-    out_opts.set_max_page_size_rows(5000);
+    out_opts.set_row_group_size_rows(num_ordered_rows);
```

Comment:
Use `constexpr` variables from header instead of magic numbers to avoid problems later on

Reply:
what problems do you expect later?

Reply:
Like if the value of `num_ordered_rows` changes in the header for some reason, which is used to generate columns, the test may fail due to magic numbers being used instead of the same constexprs

```cpp
  auto col0 = testdata::ascending<uint32_t>();
  auto col1 = testdata::descending<int64_t>();
  auto col2 = testdata::ascending<cudf::string_view>();
```